apps/src/blockly/addons/cdoFieldDropdown.js

```diff
@@ -32,6 +32,38 @@ export default class CdoFieldDropdown extends GoogleBlockly.FieldDropdown {
     }
   }
 
+  /**
+   * Sets the field's state based on the given state value. Should only be
+   * called by the serialization system.
+   *
+   * @param state The state we want to apply to the field.
+   * @override because different labs store `state` in different ways.
+   * For music lab, `state` is the value of the field.
+   * For other labs, `state` is stringified xml.
+   */
+  loadState(state) {
+    // Instead of calling on GoogleBlockly.utils.xml.textToDom(state), we do the following
+    // to avoid an error when `state` is assigned a value (music lab).
+    const parser = new DOMParser();
+    const xmlDoc = parser.parseFromString(state, 'text/xml');
+    const field = xmlDoc.querySelector('field');
+    const config = field?.getAttribute('config');
+    if (config) {
+      const value = field.textContent;
+      this.menuGenerator_ = this.getUpdatedOptionsFromConfig(config);
+      this.setValue(value);
+      return;
+    }
+    if (this.isOptionListDynamic()) {
+      this.getOptions(false);
+    }
+    if (field) {
+      this.setValue(field.textContent);
+    } else {
+      this.setValue(state); // music lab
```

Comment:
This line actually looks closest to the standard implementation in the documentation, which makes me wonder if our code should treat it as the general rule rather than the exception. Either way, I'm wondering if it makes sense to do all the xml parsing above if `state` is just the value. Is there a way we could check at the top of this function which type of `state` we have and only parse the xml if necessary?

Reply:
Sure. I can first check to see if `state` is not a stringified xml (checking first character). How does that sound?